docs/translating.md

```diff
@@ -84,12 +85,85 @@ The second step is to upload the translatable strings to Transifex using
 The final step is to get the translated files from Transifex using
 `tx pull -a`.
 
+### Marking Strings for Translation
+When you run `python manage.py makemessages`, Zulip parses all the code
+for strings which are marked for translation and dumps these strings in
+a translation resource which is uploaded to the Transifex. Zulip provides
+syntax for marking strings in both Javascript files and Handlebar
+templates. To mark strings in Javascript use:
+
+```
+i18n.t('English Text', context);
+i18n.t('English text with a __variable__', {'variable': 'Variable value'});
+```
+
+Note the syntax for declaring a variable, we enclose the variables with
+`__` instead of `{{ ... }}` because we need to differentiate the variables
+from Handlebar tags. This setting is called interpolation and you can find
+the code in `static/js/src/main.js`.
+
+To support plurals make sure your resource file contatins the related keys:
+
+```
+{
+    "en": {
+        "translation": {
+            "key": "item",
+            "key_plural": "items",
+            "keyWithCount": "__count__ item",
+            "keyWithCount_plural": "__count__ items"
+        }
+    }
+}
```

Comment:
I think there are two ways in which we can do this, one is to create a new syntax to make the parser know which translation strings should have plurals, second is to guess from the content of the strings.